### ISSUE TITLE
Restrict Default self-hosted Runner Labels

### ISSUE THREAD

**Issue by Kerruba** [rhysd/actionlint#478]
Currently, it is possible to add custom self-hosted runner labels to the runner-label checker using the following configuration:
```yaml
self-hosted-runner:
    labels:
      - private-runner-label1
      - private-runner-label2
```
By default, the self-hosted label is considered a valid option for runners.

However, when using the [Actions Runner Controller (ARC)](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners-with-actions-runner-controller/quickstart-for-actions-runner-controller) to manage self-hosted runners on Kubernetes, it is not possible to assign multiple labels to the runner-set. Additionally, the default self-hosted label is not automatically added to the runner group in this setup.

As a result, the default self-hosted label often becomes an invalid option in such scenarios.

Proposal

It would be highly beneficial to provide an option in the configuration to restrict valid self-hosted runner labels to only those explicitly defined by the user.

For example, a new field could be introduced to disable the default acceptance of the self-hosted label unless explicitly included in the user-provided list.

This feature would improve compatibility with ARC and similar setups, ensuring labels are validated based on the user’s intended configuration.

**Comment by wyardley** [comment]
An interesting additional question is how things _should_ behave with these third party hosted runner services like [Blacksmith](https://www.blacksmith.sh/), [Warpbuild](https://www.warpbuild.com), etc.

So far, I've been explicitly adding all of their versions explicitly under `self-hosted-runner.labels`, but not sure if there should be some builtin config presets for these, or a different config option?
